drivers/sensor/mpu6050/mpu6050.c

```diff
@@ -40,11 +40,25 @@ static void mpu6050_convert_gyro(struct sensor_value *val, int16_t raw_val,
 }
 
 /* see "Temperature Measurement" section from register map description */
-static inline void mpu6050_convert_temp(struct sensor_value *val,
+static inline void mpu6050_convert_temp(enum mpu6050_device_type device_type,
+					struct sensor_value *val,
 					int16_t raw_val)
 {
-	val->val1 = raw_val / 340 + 36;
-	val->val2 = ((int64_t)(raw_val % 340) * 1000000) / 340 + 530000;
+	switch (device_type) {
+	case DEVICE_TYPE_MPU6500:
+	{
+		val->val1 = raw_val / 334 + 21;
+		val->val2 = ((int64_t)(raw_val % 334) * 1000000) / 334 + 870000;
+		break;
+	}
+
+	case DEVICE_TYPE_MPU6050:
+	default:
+	{
+		val->val1 = raw_val / 340 + 36;
+		val->val2 = ((int64_t)(raw_val % 340) * 1000000) / 340 + 530000;
```

Comment:
I've done a quick check and the code gives the wrong result for negative numbers (the original is wrong, not your changes).
I think it would be easier to understand if this was refactored to create an intermediate value instead.

```suggestion
		int64_t value = (((int64_t)raw_val * 1000000) / 340) + 36530000;
		val->val1 = value / 1000000;
		val->val2 = value % 1000000;
```